sopel/irc/__init__.py

```diff
@@ -153,6 +153,8 @@ def get_irc_backend(self):
             if has_ssl:
                 backend_class = SSLAsynchatBackend
                 backend_kwargs.update({
+                    'certfile': self.settings.core.sasl_cert_file,
```

Comment:
It feels a bit wrong to have a "sasl" config related to a socket connection config.